rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java

```diff
@@ -1331,7 +1331,7 @@ public BooleanWrapper createNodeSource(String nodeSourceName, String infrastruct
             return createNodeSource(nodeSourceData, nodesRecoverable);
         } catch (RuntimeException ex) {
             logger.error(ex.getMessage(), ex);
-            if (added) {
+            if (added && !(ex instanceof NodeSourceNameAlreadyExistException)) {
```

Comment:
Why not catch NodeSourceNameAlreadyExistException instead of checking type ?

Reply:
yes. I'll change that